src/api/list.js

```diff
@@ -12,64 +13,73 @@ export default {
   },
   createItem(payload, params) {
     const url = `/api/item${objToQuery(params)}`
-    return fetch(url, {
-      method: 'POST',
-      headers: {
-        'content-type': 'application/json'
-      },
-      credentials: 'include',
-      body: JSON.stringify(payload)
-    }).then(response => response.json())
+    return http
+      .fetch(url, {
+        method: 'POST',
+        headers: {
+          'content-type': 'application/json'
+        },
+        credentials: 'include',
+        body: JSON.stringify(payload)
+      })
+      .then(response => response.json())
   },
   listUnrankedItems(params) {
     const url = `/api/unranked_items${objToQuery(params)}`
-    return fetch(url, {
-      method: 'GET',
-      headers: {
-        'content-type': 'application/json'
-      },
-      credentials: 'include'
-    }).then(response => response.json())
+    return http
+      .fetch(url, {
+        method: 'GET',
+        headers: {
+          'content-type': 'application/json'
+        },
+        credentials: 'include'
+      })
+      .then(response => response.json())
   },
   listRankedItems(params) {
     const url = `/api/user_ranking${objToQuery(params)}`
-    return fetch(url, {
-      method: 'GET',
-      headers: {
-        'content-type': 'application/json'
-      },
-      credentials: 'include'
-    }).then(response => response.json())
+    return http
+      .fetch(url, {
+        method: 'GET',
+        headers: {
+          'content-type': 'application/json'
+        },
+        credentials: 'include'
+      })
+      .then(response => response.json())
   },
   listAllLists() {
     const url = `/api/lists`
-    return fetch(url, {
-      method: 'GET',
-      headers: {
-        'content-type': 'application/json'
-      },
-      credentials: 'include'
-    })
+    return http
+      .fetch(url, {
+        method: 'GET',
+        headers: {
+          'content-type': 'application/json'
+        },
+        credentials: 'include'
+      })
       .then(response => response.json())
       .then(lists => lists.filter(list => !!list.name_id))
   },
   postItemRank(previousItemId, targetItemId, params) {
     const url = `/api/user_ranking${objToQuery(params)}`
-    return fetch(url, {
-      method: 'POST',
-      headers: {
-        'content-type': 'application/json'
-      },
-      credentials: 'include',
-      body: JSON.stringify({
-        prev_item_id: previousItemId,
-        item_id: targetItemId
+    return http
+      .fetch(url, {
+        method: 'POST',
+        headers: {
+          'content-type': 'application/json'
+        },
+        credentials: 'include',
+        body: JSON.stringify({
+          prev_item_id: previousItemId,
+          item_id: targetItemId
+        })
       })
-    }).then(response => response.json())
+      .then(response => response.json())
```

Comment:
We don't need it really, but I'd suggest we make a decorated `http.fetch` that does this `response.json()` call...seems easy enough to forget to do.

Reply:
I don't think it'd be a stretch to include the common options stuff, I think 
```        
headers: {
  'content-type': 'application/json'
},
credentials: 'include'
```

Reply:
Good ideas!  I think I'd like to handle all of this auth stuff in a [separate PR](https://github.com/SharpNotions/pantry-list-frontend/pull/39) but I'll incorporate this feedback there.